fix(gnovm): gno test dump-machine-state && limit string size for render

ltzmaxwell opened the pull request:
- limit size of string rendering
- add a dump-machine-state flag for: gno test -dump-machine-state.

ltzmaxwell committed:
fixup

ltzmaxwell committed:
set proper allocation upper limit

ltzmaxwell committed:
Merge remote-tracking branch 'main/master' into fix/maxwell/gg

ltzmaxwell committed:
rm unused

ltzmaxwell committed:
ajust const

ltzmaxwell committed:
Merge remote-tracking branch 'main/master' into fix/maxwell/gg2

ltzmaxwell committed:
restore const; no print m.String()

ltzmaxwell committed:
fix test

ltzmaxwell committed:
debug

Gno2D2 commented:
#### 🛠 PR Checks Summary
 🔴 Changes related to gnoweb must be reviewed by its codeowners


##### Manual Checks (for Reviewers):
- [ ] **IGNORE** the bot requirements for this PR (force green CI check)
- [ ] Determine if infra needs to be updated before merging


<details><summary>Read More</summary>

🤖 This bot helps streamline PR reviews by verifying automated checks and providing guidance for contributors and reviewers.

##### ✅ Automated Checks (for Contributors):
 🟢 Maintainers must be able to edit this pull request ([more info](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork))
 🔴 Changes related to gnoweb must be reviewed by its codeowners


##### ☑️ Contributor Actions:
1. Fix any issues flagged by automated checks.
2. Follow the Contributor Checklist to ensure your PR is ready for review.
	- Add new tests, or document why they are unnecessary.
	- Provide clear examples/screenshots, if necessary.
	- Update documentation, if required.
	- Ensure no breaking changes, or include `BREAKING CHANGE` notes.
	- Link related issues/PRs, where applicable.

##### ☑️ Reviewer Actions:
1. Complete manual checks for the PR, including the guidelines and additional checks if applicable.

##### 📚 Resources:
- [Report a bug with the bot](https://www.github.com/gnolang/gno/issues/3238).
- [View the bot’s configuration file](https://github.com/gnolang/gno/tree/master/contribs/github-bot/internal/config/config.go).

<details><summary><b>Debug</b></summary><blockquote>
<details><summary><b>Automated Checks</b></summary><blockquote>

<details><summary>Maintainers must be able to edit this pull request (more info)</summary><blockquote>

### If
```
🟢 Condition met
└── 🟢 And
    ├── 🟢 The base branch matches this pattern: ^master$
    └── 🟢 The pull request was created from a fork (head branch repo: ltzmaxwell/gno)

```
### Then
```
🟢 Requirement satisfied
└── 🟢 Maintainer can modify this pull request

```
</blockquote></details>

<details><summary>Changes related to gnoweb must be reviewed by its codeowners</summary><blockquote>

### If
```
🟢 Condition met
└── 🟢 And
    ├── 🟢 The base branch matches this pattern: ^master$
    └── 🟢 A changed file matches this pattern: ^gno.land/pkg/gnoweb/ (filename: gno.land/pkg/gnoweb/frontend/css/05-composition.css)

```
### Then
```
🔴 Requirement not satisfied
└── 🔴 Or
    ├── 🔴 Or
    │   ├── 🔴 And
    │   │   ├── 🔴 Pull request author is user: alexiscolin
    │   │   └── 🔴 This user reviewed pull request: gfanton (with state "APPROVED")
    │   └── 🔴 And
    │       ├── 🔴 Pull request author is user: gfanton
    │       └── 🔴 This user reviewed pull request: alexiscolin (with state "APPROVED")
    └── 🔴 And
        ├── 🟢 Not (🔴 Pull request author is user: alexiscolin)
        ├── 🟢 Not (🔴 Pull request author is user: gfanton)
        └── 🔴 Or
            ├── 🔴 This user reviewed pull request: alexiscolin (with state "APPROVED")
            └── 🔴 This user reviewed pull request: gfanton (with state "APPROVED")

```
</blockquote></details>

</blockquote></details>


<details><summary><b>Manual Checks</b></summary><blockquote>

<details><summary>**IGNORE** the bot requirements for this PR (force green CI check)</summary><blockquote>

### If
```
🟢 Condition met
└── 🟢 On every pull request

```
### Can be checked by

- Any user with comment edit permission

</blockquote></details>

<details><summary>Determine if infra needs to be updated before merging</summary><blockquote>

### If
```
🟢 Condition met
└── 🟢 And
    ├── 🟢 The base branch matches this pattern: ^master$
    └── 🟢 Or
        ├── 🔴 A changed file matches this pattern: Dockerfile
        ├── 🟢 A changed file matches this pattern: ^misc/deployments (filename: misc/deployments/test10.gno.land/CHECKLIST.md)
        ├── 🔴 A changed file matches this pattern: ^misc/docker-
        └── 🔴 A changed file matches this pattern: ^.github/workflows/release.*\.yml$

```
### Can be checked by
 - team devops

</blockquote></details>

</blockquote></details>

</blockquote></details>

</details>

ltzmaxwell committed:
cleanup

ltzmaxwell committed:
cleanup

ltzmaxwell committed:
cleanup

ltzmaxwell committed:
debug

ltzmaxwell committed:
fixup

jaekwon committed:
Create tictac.md

Do not remove this file.
It is important information, and a canary file.

jaekwon committed:
Added file that proves Reddit censors truthful physics discoveries.

Of all places, /r/UFOResearch should not censor this kind of information.
DO NOT DELETE THIS FILE, DO NOT TRUST ANYONE WHO DELETES THIS FILE.

jaekwon committed:
Update tictac.md

added law of spinning mass.

moul committed:
fix(params): SetBytes should not alter the value bytes (alternative) (#4926)

Alternative for #4905 

Kept:
- Removed GetRaw/SetRaw since they were unused.
- SetBytes to use directly the store, preventing the value to be
aminoJSON encoded.

Updated:
- Preserve API consistency.

---------

Signed-off-by: moul <94029+moul@users.noreply.github.com>
Co-authored-by: Thomas Bruyelle <thomas.bruyelle@tendermint.com>

jeronimoalbi committed:
feat(examples): simple Markdown forms package (#4977)

Package aims to be a very simple package to define Gno-Flavored Markdown
forms.

It should be easy to update when new form elements or attributes are
introduced.

---------

Co-authored-by: Antoine Eddi <5222525+aeddi@users.noreply.github.com>

thehowl committed:
fix(vm/keeper): pre-load stdlib type check cache (#4925)

longer term fix for #4874, re-introducing the type check cache for
stdlibs.

jeronimoalbi committed:
feat(boards2): change realm to use `gno.land/p/gnoland/boards` (#4939)

PR changes Boards2 realm to use `gno.land/p/gnoland/boards` package
which makes the realm leaner.

It also improves `BasicPermissions` implementation by using a custom
members storage implementation that makes users iteration cheaper.

mvallenet committed:
fix(vm/keeper): improve error for non crossing msg call error (#4899)

fix #4432

---------

Co-authored-by: Morgan <morgan@morganbaz.com>

moul committed:
test: add a banker security txtar (#4987)

Signed-off-by: moul <94029+moul@users.noreply.github.com>

---------

Signed-off-by: moul <94029+moul@users.noreply.github.com>
Co-authored-by: Antoine Eddi <5222525+aeddi@users.noreply.github.com>

jeronimoalbi committed:
feat(boards2): open boards support (#4993)

jeronimoalbi committed:
feat(boards2): support Markdown forms (#4996)

jeronimoalbi committed:
chore(boards2): enable open boards field in create board form (#4998)

Also add a couple of minor fixes

jaekwon committed:
first commit of whitepaper; draft; DO NOT SHARE, deadman switch

moonia committed:
fix(gnoweb): handle multiline value for gno-textarea in forms (#4978)

Closes https://github.com/gnolang/gno/issues/4968

## Description

The `<gno-textarea>` component was literally displaying the `\n`
sequence. After fix, the parser now interprets the `\n` sequence as a
real newline character.

### Example

```
<gno-form>
  <gno-textarea name="content" value="         First line\nSecond line" />
</gno-form>
```

### Before fix: 
<img width="931" height="236" alt="Capture d’écran 2025-12-15 à 23 22
23"
src="https://github.com/user-attachments/assets/220f85ee-c5ad-4dd3-9b26-863bd84e6a58"
/>


### After fix: 
<img width="945" height="239" alt="Capture d’écran 2025-12-15 à 23 34
27"
src="https://github.com/user-attachments/assets/e8f7551e-749f-4259-ba50-5dae333cf487"
/>

---------

Co-authored-by: alexiscolin <alexis@jaunebleu.co>

moonia committed:
feat(gnoweb): support required attribute for markdown form fields (#4974)

## Description 
Closes https://github.com/gnolang/gno/issues/4970
This PR adds support for the required="true" attribute in Gno-flavored
markdown form fields.

### Usage 

- `required="true"` can be added to: `<gno-input>`, `<gno-textarea>`,
`<gno-select>`
- Required fields: display a * indicator next to the label

```
<gno-form>
    <gno-textarea name="bio" required="true" />
</gno-form>
```
### Preview
<img width="936" height="238" alt="Capture d’écran 2025-12-15 à 16 09
19"
src="https://github.com/user-attachments/assets/3cd743a8-350c-4906-b33f-25fb5ecb9ac8"
/>

---------

Co-authored-by: Alexis Colin <alexis@jaunebleu.co>

thehowl committed:
feat(github-bot): add MikaelVallenet to review team (#5036)

davd-gzl committed:
feat(valopers): add server type support for validator (#5015)

Co-authored-by: Miloš Živković <milos.zivkovic@tendermint.com>

mvallenet committed:
docs(stdlibs): fix outdated reference to the testing package (#4929)

Co-authored-by: Miloš Živković <milos.zivkovic@tendermint.com>

alexiscolin committed:
fix(boards2): improve ux / ui (#5022)

This PR enhances the boards2 realm UI with improved navigation, cleaner
post layout, and better visual hierarchy.

Related to https://github.com/gnolang/gno/issues/4999

## Changes
### Navigation Improvements

- Clickable breadcrumbs: "Boards" in headers now links back to the
boards list
- Back links on forms: Added "← Back to thread" and "← Back to board"
links on all form pages (Create Thread, Edit Thread, Repost Thread, Edit
Comment, Reply, Flag)
- Thread page breadcrumb: Shows [Boards] › [BoardName] with both
elements clickable to navigate into the realm from the realm

### Post Layout
- Author info at top: Moved author name and date to the beginning of
posts instead of the end to improve readability
- Format: **@user** · date #postID appears before the content
- Action links with arrow: Added ↳ prefix before action links (Flag •
Reply • Edit • Delete) to make it more obvious (as set of actions)
- Add a reply and repost counter to quickly identify popular posts and
trends.
- Remove the sidebar when the table of contents is empty.
- Reduce global spacing in the UI.

### Visual Hierarchy
- H6 headings for listings: Board names and thread titles in listings
now use H6 instead of bold text to add content to the sidebar.
- Roles badges have been added to enhance transparency and help users
identify who is speaking when messages pertain to rules.

### Forms
- Implements the `required` attributes on form fields to submit or
display an error when fields are empty or overlooked.
- A `Back to Previous Page` button is placed under the form to help
users understand that they can return after completing the form.

## Screenshots

<img width="1463" height="585" alt="Capture d’écran 2026-01-07 à 16 50
36"
src="https://github.com/user-attachments/assets/47477e49-45b4-4bc1-82b9-7751daa0228d"
/>
<img width="1464" height="982" alt="Capture d’écran 2026-01-07 à 16 50
55"
src="https://github.com/user-attachments/assets/04deb551-0472-43f9-ab1d-bb068369028b"
/>
<img width="1532" height="875" alt="Capture d’écran 2026-01-07 à 17 47
36"
src="https://github.com/user-attachments/assets/33af19c5-1644-4d1d-971b-9119eac48ea5"
/>

<img width="1386" height="445" alt="Capture d’écran 2026-01-08 à 21 42
13"
src="https://github.com/user-attachments/assets/f9aea63d-26b2-4687-9f8e-80fb105fdf31"
/>

---------

Co-authored-by: Jerónimo Albi <894299+jeronimoalbi@users.noreply.github.com>

gfanton committed:
feat(gnoweb): add `params:changed` (#5002)

This PR adds a new `params:changed` custom event to gnoweb, enabling
secure parameter update.

cc @jinoosss

---------

Co-authored-by: Alexis Colin <alexis@jaunebleu.co>

sw360cab committed:
chore(staging): staging env infra setup improvements (#5034)

Improved these items:
- Traefik config to get more precise data in logs
- Traefik config to persist Letsencrypt folder and avoid regenerating
each restart
- Gnoweb performances by adding Rate Limiter Traefik Middleware

moonia committed:
docs: fix typo errors across documentation (#5012)

moonia committed:
fix(docs): align docs with testnet test10 (#5021)

Co-authored-by: Nemanja Aleksic <kouteki@gmail.com>

zivkovicmilos committed:
feat: add initial `test10` deployment (#5045)

## Description

This PR adds the test10 deployment to the monorepo.

ltzmaxwell committed:
cleanup

ltzmaxwell committed:
Merge remote-tracking branch 'main/master' into fix/maxwell/machine_state_print

ltzmaxwell committed:
fixup

ltzmaxwell committed:
fixup

ltzmaxwell committed:
add test

thehowl commented:
needs rebase
